aten/src/ATen/core/ivalue.h

```diff
@@ -372,12 +372,7 @@ struct TORCH_API IValue final {
     return payload.as_tensor.unsafeGetTensorImpl();
   }
 
-  IValue(at::Storage s) : tag(Tag::Storage), is_intrusive_ptr(static_cast<bool>(s)) {
-    // Note: the undefined tensor is not refcounted, so while it
-    // is tagged as a tensor, is_intrusive_ptr is set to false.
-    // This is not an optional optimization: our incref call
-    // *will not* do the right thing when called on an
-    // undefined tensor.
+  IValue(at::Storage s) : tag(Tag::Storage), is_intrusive_ptr(true) {
```

Comment:
Could you clarify this change? thanks

Reply:
the behavior for these types was out of sync with other types, and the comment about refcounting was out of date. Making this change allows removing `is_intrusive_ptr`, which is a deterministic function of the tag, in a follow-up.

That said, it doesn't look like this change is necessary to make the new tests in this diff pass, so I'll split it out.